opm/parser/eclipse/EclipseState/EclipseState.hpp

```diff
@@ -123,7 +125,7 @@ namespace Opm {
 
 
         const UnitSystem& m_deckUnitSystem;
-        const ParseContext& m_parseContext;
+        const ParseContext m_parseContext;
```

Comment:
> But if they're to be copied anyway you can't avoid this cost, regardless of reference passing. If a copy isn't needed then the argument should obviously be a reference.

In my personal opinion, objects of non-primitive types should always passed as const references and if they're required after the argument object has potentialy gone out of scope, the called function is responsible for copying it. some people (@atgeirr!?) are even more strict here and require even primitive by-value objects to be marked const.

Anyway, if you and @joakim-hove prefer the by-value call here, I'm certainly not going to object...